src/Razor/src/Microsoft.AspNetCore.Razor.Language/Intermediate/MarkupElementIntermediateNode.cs

```diff
@@ -20,6 +20,7 @@ public sealed class MarkupElementIntermediateNode : IntermediateNode
         {
             return
                 c as HtmlAttributeIntermediateNode == null &&
+                c as SplatIntermediateNode == null &&
```

Comment:
@ajaybhargavb this is an artifact of our IR bits not having a well-defined start/end tag representation.

Reply:
Why do we need splat here yet we don't have the corresponding component attribute IR node here?

Reply:
A component attribute will not appear as a child of a markup element.